### ISSUE TITLE
Nested collection fields

### ISSUE THREAD

**Issue by nielvrom** [sonata-project/SonataAdminBundle#3583]
I'm having problems creating my form for creating a course. This is a part of my database scheme for which I'm trying to create a form:

<img width="352" alt="schermafbeelding 2016-02-11 om 09 20 31" src="https://cloud.githubusercontent.com/assets/2797731/12971923/c01f21ba-d0a0-11e5-85bb-42a526c7cda8.png">

So which I'm trying to do is create a course where I can create sessions and dates attached to that session. It should look something like this:

<img width="252" alt="schermafbeelding 2016-02-06 om 17 13 53" src="https://cloud.githubusercontent.com/assets/2797731/12971939/e5ae3f60-d0a0-11e5-8dfc-828176bdeded.png">

In my **CourseAdmin** class I have:

``` php
$formMapper
        ->add('name',                   'text',         array('label' => 'Naam'))
        ->add('description',            'textarea',     array('label' => 'Beschrijving'))
        ->add('materials',              'textarea',     array('label' => 'Benodigde materialen'))
        ->add('numberOfParticipants',   'number',       array('label' => 'Aantal deelnembers'))
        ->add('numberOfDays',           'number',       array('label' => 'Aantal dagen'))
        ->add('price',                  'number',       array('label' => 'Prijs'))
        ->add('priceKmo',               'number',       array('label' => 'KMO-portefeuille Prijs'))

        ->add('location', 'sonata_type_model', array('expanded' => true, 'by_reference' => false, 'multiple' => true, 'btn_add' => false))

        ->add('session', 'sonata_type_collection', array(
            'by_reference' => false,
            'type_options' => array(
                // Prevents the "Delete" option from being displayed
                'delete' => false,
                'delete_options' => array(
                    // You may otherwise choose to put the field but hide it
                    'type'         => 'hidden',
                    // In that case, you need to fill in the options as well
                    'type_options' => array(
                        'mapped'   => false,
                        'required' => false,
                    )
                )
            )
        ), array(
            'edit' => 'inline',
            'inline' => 'table',
            'sortable' => 'position'
        ))
    ;
```

In my **SessionAdmin** class I have:

``` php
$formMapper
        ->add('type',      'text',     array('label' => 'Type opleiding (Dag / Avond)'))
        ->add('moment', 'sonata_type_collection', array(
            'by_reference' => false,
            'type_options' => array(
                // Prevents the "Delete" option from being displayed
                'delete' => false,
                'delete_options' => array(
                    // You may otherwise choose to put the field but hide it
                    'type'         => 'hidden',
                    // In that case, you need to fill in the options as well
                    'type_options' => array(
                        'mapped'   => false,
                        'required' => false,
                    )
                )
            )
        ), array(
            'edit' => 'inline',
            'inline' => 'table',
            'sortable' => 'position'
        ))
    ;
```

And in my **MomentAdmin** class I have:

``` php
$formMapper
        ->add('time',      'date',     array('label' => 'Datum'))
    ;
```

The problem in my form is when I try to add a date to my session I get the following error:

> FatalErrorException: Error: Call to a member function getName() on null in /myproject/app/cache/dev/classes.php line 9772

So, I can add a session but when I try to add a date to my session I'm getting the error ... .
The function causing the error is the following:

``` php
public function appendFormFieldElement(AdminInterface $admin, $subject, $elementId)
{
    $formBuilder = $admin->getFormBuilder();
    $form = $formBuilder->getForm();
    $form->setData($subject);
    $form->submit($admin->getRequest());
    $childFormBuilder = $this->getChildFormBuilder($formBuilder, $elementId);
    $fieldDescription = $admin->getFormFieldDescription($childFormBuilder->getName());

    try {
        $value = $fieldDescription->getValue($form->getData());
    } 
    catch (NoValueException $e) {
        $value = null;
    }

    $data = $admin->getRequest()->get($formBuilder->getName());

    if (!isset($data[$childFormBuilder->getName()])) {
        $data[$childFormBuilder->getName()] = array();
    }

    $objectCount = count($value);
    $postCount = count($data[$childFormBuilder->getName()]);
    $fields = array_keys($fieldDescription->getAssociationAdmin()->getFormFieldDescriptions());
    $value = array();

    foreach ($fields as $name) {
        $value[$name] ='';
    }

    while ($objectCount < $postCount) {
        $this->addNewInstance($form->getData(), $fieldDescription);
        ++$objectCount;
    }

    $this->addNewInstance($form->getData(), $fieldDescription);
    $data[$childFormBuilder->getName()][] = $value;
    $finalForm = $admin->getFormBuilder()->getForm();
    $finalForm->setData($subject);
    $finalForm->setData($form->getData());

    return array($fieldDescription, $finalForm);
}
```

And the rule that's causing the error is :

`$fieldDescription = $admin->getFormFieldDescription($childFormBuilder->getName());`

**Comment by efiorello** [comment]
I think your problem is caused by not having created the method public function __toString() in your entity.

**Comment by nielvrom** [comment]
All my entities have a __toString() function ... .

**Comment by chalasr** [comment]
Possible duplicate of #262, #1228, #1327 and #1971 .

**Comment by jordisala1991** [comment]
Closing, as @chalasr said (thanks btw), it was fixed on #3553
